src/extlibs/er_force_sim/src/amun/simulator/simrobot.cpp

```diff
@@ -421,7 +421,12 @@ void SimRobot::begin(SimBall *ball, double time)
 
         if (m_sslCommand.kick_angle() == 0)
         {
-            power = qBound(0.05f, m_sslCommand.kick_speed(), m_specs.shot_linear_max());
+            // we subtract the current speed of the ball from the intended kick speed
+            // this ensures the ball leaves the robot at exactly the speed we want
+            power = qBound(
+                0.05f,
+                m_sslCommand.kick_speed() - (ball->speed().length() / SIMULATOR_SCALE),
+                m_specs.shot_linear_max());
```

Comment:
on saturday, we talked about doing this bound only if the ball's current speed is lower than some threshold